lib/auth/access.go

```diff
@@ -37,7 +37,7 @@ func (a *Server) CreateRole(ctx context.Context, role types.Role) (types.Role, e
 		return nil, trace.Wrap(err)
 	}
 
-	if err := a.emitter.EmitAuditEvent(a.closeCtx, &apievents.RoleCreate{
+	if err := a.emitter.EmitAuditEvent(apievents.AuditContext{Context: a.closeCtx}, &apievents.RoleCreate{
```

Comment:
Why do we have some creation using the constructor and some creating the instance directly?